content/developers/internals/data-model/usage-example.en.md

```diff
@@ -0,0 +1,273 @@
+
+If two track sections are not connected by a `TrackSectionLink`, a train in an *OSRD* simulation cannot go from the one to the other.
+
+In our example, creating the links is straight-forward: since we chose to have long track sections, we only have to link them at switches. Happily, in our generation scripts, like in [small_infra.py](https://github.com/DGEXSolutions/osrd/blob/dev/core/examples/generated/scripts/small_infra.py), when creating switches, the corresponding `TrackSectionLink`s are automatically created.
+
+Let's delve a bit on the switches.
+
+### Switches
+
+<!-- TODO: what it is -->
+The switches are the parts needed to link more than two different track sections together.
+
+A switch manages the different links that are associated to it (always at least two). In general, it translates into the following rule: when one link is used, the others cannot be used at the same time. This is because if two links were used by trains at the same time, it would be an accident.
+
+Since the kinds of links that we need to create at a switch often happen to be redundant, we have `SwitchType` objects, which gather what links should be created, when creating this particular type of switch. We will describe this object in the next subsection.
+
+<!-- TODO: how it's implemented -->
+<!-- TODO: how we use it -->
+
+#### Switch Types
+
+A switch type contains two attributes:
+
+* `ports`: A list of names, that will be used to reference the ports. It is best described by example (see next paragraph).
+* `groups`: A mapping from group names to lists of connections between two ports. When one group is active, only the connections in its list are active.
+
+In *OSRD*, we currently use three switch types.
+
+**1) Point Switch**
+
+This is the most standard kind of switch and can be conceived as two tracks merging into one, or one track splitting into two.
+
+This switch type has three ports: *base*, *left* and *right*.
+
+![Point switch schema](../svg_schemas/point_switch.en.svg)
+
+There are two groups, each with one connection in their list: `LEFT`, which links *base* to *left*, and `RIGHT` which links *base* to *right*.
+
+Thus, at any given moment, a train can be able to go from *base* to *left* or from *base* to *right* but never to both at the same time. A train cannot go from *left* to *right*.
+
+A Point Switch only has two positions:
+
+![Point switch positions schema](../svg_schemas/point_switch_positions.en.svg)
+
+**2) Cross Switch**
+
+This is simply two tracks crossing each other.
+
+This type has four ports: *north*, *south*, *east* and *west*.
+
+![Cross switch schema](../svg_schemas/cross_switch.en.svg)
+
+It has two groups containing each one connection: *north* to *south* and *west* to *east*. Thus the cross switch has two positions, as shown here:
+
+![Cross switch positions schema](../svg_schemas/cross_switch_positions.en.svg)
```

Comment:
A cross switch has only one group:
```json
    {
        "id": "cross",
        "ports": [
            "north",
            "south",
            "east",
            "west"
        ],
        "groups": {
            "static": [
                {
                    "dst": "west",
                    "src": "east",
                    "bidirectional": true
                },
                {
                    "dst": "south",
                    "src": "north",
                    "bidirectional": true
                }
            ]
        }
    }
```

Reply:
@multun told me that this type of switch should have two groups and not one, so I chose to document the "normal" situation, but I can document the one group documentation if your prefer. However, first I need to ask you questions about how groups work because then I did not understand it correctly.

Reply:
We can check with @loic-hamelin or @nicolaswurtz but I am pretty sure that this type of switch is passive.